app/controllers/students_groups_controller.rb

```diff
@@ -0,0 +1,32 @@
+class StudentsGroupsController < ApplicationController
+  def new
+     @student = Student.find(params[:student_id])
+     @students_group = StudentsGroup.new
+     authorize @students_group
+   end
+
+  def create
+    @student = Student.find(params[:student_id])
+    @students_group = StudentsGroup.new(students_group_params)
+    @students_group.student = @student
+    authorize @students_group
+    if @students_group.save
+      redirect_to student_path(@student)
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @students_group = StudentsGroup.find(params[:id])
+    @students_group.destroy
```

Comment:
first authorize it, only then destroy it! (else you can destroy stuff you're not supposed to)

Reply:
right, changed that so the line 21 has authorize function

Reply:
good, but you need to find it before authorizing it ;P this way he still doesn't know what @students_group is when you call pundit